pkg/controller/stacks/stack/stack.go

```diff
@@ -86,6 +86,11 @@ var (
 	}
 
 	disableAutoMount = false
+
+	allowPrivilegeEscalation       = false
+	privileged                     = false
+	runAsNonRoot                   = true
+	runAsUser                int64 = 1000
```

Comment:
what is user 1000? is that standard k8s practice? a comment may be helpful for maintaining this code

edit: ah, it's called out in the design doc.  maybe a quick mention of that idea here in the code too?

Reply:
Definitely will add a comment :+1: I do not know if this is typical k8s practice, mostly because I don't know of too many systems that are doing what we are doing with stacks. I know that running a container as `USER 1000` is a common practice, but typically you would know the container you are running instead of an arbitrary one in this case. @lukeweber or @negz any thoughts on this?

Reply:
Without having any background on this idea, It "feels" a bit brittle and non-portable to use what seems to me as a random user ID.  From searching around for more information on it this morning, it seems like it's commonly suggested, but it's harder to find a good justification for doing so.  It almost feels like one of those "examples" that becomes more idiomatic through copy/paste in the ecosystem 😉 

This article gets into a bit more description, in its "Do Not Run As Root" section: https://www.bluematador.com/blog/kubernetes-security-essentials

What happens if there is no user 1000 in the uid namespace the container is running in? Is 1000 always guaranteed to exist, or does that not actually matter?

This article indicates you can run a container as a totally random user ID and in fact that's what OpenShift does, see section "Use Case #2 – Running API Simulator Container Using Random Non-root User":
https://apisimulator.io/run-docker-containers-non-root-users-random-user-ids/

So while to me this seemed non-intuitive or a bit fragile, perhaps this is a fairly acceptable approach to "force" the container to not run as root, which in many cases it will try to do, for instance when no specific user was specified in the container build process.  Any other thoughts?

Reply:
This pretty much sums up my thoughts as well. I also don't love setting to a user ID explicitly for a general "installer". However, we are setting to a user ID rather than a user, so we do not need to be concerned about the user not existing.

We could alternatively say that the onus falls on the creator of the Docker image to ensure that it runs as non-root, something that we should do in Crossplane-maintained images anyway. This might make it a bit easier to debug as the container will just fail rather than potentially being run without root permissions that it requires for some reason. Basically what we are trying to do here is make it such that installations do not fail for mis-configured Docker images, which I don't know if this is something we should do or not. I could go either way here.

Reply:
For posterity: decision here is to not set user and document that containers must be able to run as non-root.